jsf/subsystem/src/test/java/org/jboss/as/jsf/deployment/JSFModuleIdFactoryTestCase.java

```diff
@@ -52,25 +54,44 @@ public void validSlotTest() {
     @Test
     @Ignore("Depends on https://issues.redhat.com/browse/WFLY-17405")
     public void modIdsTest() {
-        Assert.assertEquals(API_MODULE, factory.getApiModId("main").getName());
-        Assert.assertEquals("main", factory.getApiModId("main").getSlot());
-        Assert.assertEquals(IMPL_MODULE, factory.getImplModId("main").getName());
-        Assert.assertEquals("main", factory.getImplModId("main").getSlot());
-        Assert.assertEquals(INJECTION_MODULE, factory.getInjectionModId("main").getName());
-        Assert.assertEquals("main", factory.getInjectionModId("main").getSlot());
-
-        Assert.assertEquals(API_MODULE, factory.getApiModId("myfaces").getName());
-        Assert.assertEquals("myfaces", factory.getApiModId("myfaces").getSlot());
-        Assert.assertEquals(IMPL_MODULE, factory.getImplModId("myfaces").getName());
-        Assert.assertEquals("myfaces", factory.getImplModId("myfaces").getSlot());
-        Assert.assertEquals(INJECTION_MODULE, factory.getInjectionModId("myfaces").getName());
-        Assert.assertEquals("myfaces", factory.getInjectionModId("myfaces").getSlot());
-
-        Assert.assertEquals(API_MODULE, factory.getApiModId("myfaces2").getName());
-        Assert.assertEquals("myfaces2", factory.getApiModId("myfaces2").getSlot());
-        Assert.assertEquals(IMPL_MODULE, factory.getImplModId("myfaces2").getName());
-        Assert.assertEquals("myfaces2", factory.getImplModId("myfaces2").getSlot());
-        Assert.assertEquals(INJECTION_MODULE, factory.getInjectionModId("myfaces2").getName());
-        Assert.assertEquals("myfaces2", factory.getInjectionModId("myfaces2").getSlot());
+        // Canonicalize module names using ModuleIdentifierUtil
+        String mainCanonical = canonicalModuleIdentifier("main", "main");
+        String myfacesCanonical = canonicalModuleIdentifier("myfaces", "myfaces");
+        String myfaces2Canonical = canonicalModuleIdentifier("myfaces2", "myfaces2");
+
+        // Test API module for "main"
+        String apiModIdMain = factory.getApiModId(mainCanonical);
+        String implModIdMain = factory.getImplModId(mainCanonical);
+        String injectionModIdMain = factory.getInjectionModId(mainCanonical);
```

Comment:
Just pass "main" as the param for these.

The canonicalModuleIdentifier code does escaping of inputs it gets, but none of the inputs in this test need escaping, so finding ways to invoke it isn't necessary and adds complexity. The test author can rightly expect to use "main", "myfaces", "myfaces2" as test data and to get the expected result.